src/sqlparser/src/parser.rs

```diff
@@ -236,6 +245,25 @@ impl Parser {
         Ok(Statement::Analyze { table_name })
     }
 
+    /// Check is enter array expression.
+    pub fn peek_array_depth(&self) -> i32 {
+        self.array_depth
+    }
+
+    /// When enter specify ARRAY prefix expression.
+    pub fn increase_array_depth(&mut self, num: i32) {
+        self.array_depth += num;
+    }
+
+    /// When exit specify ARRAY prefix expression.
+    pub fn decrease_array_depth(&mut self, num: i32) {
+        self.array_depth -= num;
+    }
+
+    pub fn is_in_named_array(&self) -> bool {
```

Comment:
The name is somewhat confusing now. The outermost array must be named and we can just call it `is_in_array`.

Reply:
Okay.